### ISSUE TITLE
Tokenizer's encoder behavior is different between huggingface and onnxgenai and that causes slight model accuracy drop

### ISSUE THREAD

**Issue by fujikosu** [microsoft/onnxruntime-genai#1248]
**Describe the bug**
I got finetuned phi3.5 mini model and I converted it to onnx by this library. Compared to huggingface's pipeline, onnxruntime-genai doesn't apply phi chat template automatically or doesn't provide a way to do so in the package, so I adopted apply_chat_template from huggingface and was doing tokenization like below.

```py
formatted_text = hf_tokenizer.apply_chat_template(messages, tokenize=False)
og.Tokenizer(model).encode(formatted_text)
```

But I realized that input tokens I get from this operation is slightly different from the ones from huggingface pipeline and that was causing slightly different behavior of the model.

This is diff analysis between tokens from onnx (on the left) and tokens from huggingface (on the right)
![Image](https://github.com/user-attachments/assets/8e2c398f-a8c8-438d-a412-f70e1cf80ab3)

To exactly match with huggingface tokenization result, I needed to implement this manual format instead

```py
phi_chat_template = "<|system|>{system}<|end|><|user|> {user}<|end|><|endoftext|>"
```

digging in deeper, I found diff is coming from the fact that \n is trimmed after each special token like <|system|> in huggingface and not in onnx.

And that is because I think huggingface tokenization respects this `"rstrip": true` setting in tokenizer_config.json and onnxruntime-genai doesn't use this information

```json
    "32006": {
      "content": "<|system|>",
      "lstrip": false,
      "normalized": false,
      "rstrip": true,
      "single_word": false,
      "special": true
    },
    "32007": {
      "content": "<|end|>",
      "lstrip": false,
      "normalized": false,
      "rstrip": true,
      "single_word": false,
      "special": true
    },
    "32010": {
      "content": "<|user|>",
      "lstrip": false,
      "normalized": false,
      "rstrip": true,
      "single_word": false,
      "special": true
    },
```

Lots of people come from pretrained torch models or their own finetuned torch models that were trained with tokenization in huggingface to onnx and this tokenization behavior gap might cause unexpected model behavior change (in our case, there was slight but noticeable accuracy drop on the metric) so it'd be great if this was fixed or documented well at least

**To Reproduce**

huggingface tokenization

```py
simple = {
    "messages": [
        {
            "role": "system",
            "content": "test1",
        },
        {"role": "user", "content": "test2"},
    ]
}
hf_tokenizer = AutoTokenizer.from_pretrained(model_path)
input_tokens = hf_tokenizer.apply_chat_template(messages, tokenize=True)
```

onnxruntime-genai tokenization

```py
simple = {
    "messages": [
        {
            "role": "system",
            "content": "test1",
        },
        {"role": "user", "content": "test2"},
    ]
}
hf_tokenizer = AutoTokenizer.from_pretrained(model_path)
onnx_tokenizer = og.Tokenizer(model_path)
formatted_text = hf_tokenizer.apply_chat_template(messages, tokenize=False)
input_tokens = onnx_tokenizer.encode(formatted_text)
```

**Expected behavior**
onnxgenai's tokenizatiton works exactly at the same way as huggingface's tokenization that respects tokenizer_config.json's information

**Screenshots**

**Desktop (please complete the following information):**
 - OS: linux (gpu container)
 - Version: Ubuntu 22.04.3 LTS

**Additional context**

**Comment by sayanshaw24** [comment]
I believe this issue was fixed by this recent PR https://github.com/microsoft/onnxruntime-extensions/pull/887. Once onnxruntime-genai updates to the latest ORT Extensions version (which it uses for pre- and post-processing) this should resolve the problem. @kunal-vaishnavi @baijumeswani could you close this issue when the latest changes are pulled in for the next release?

**Comment by baijumeswani** [comment]
We will include the changes in the next ort-genai release (0.7.0). I'll close this issue now since the PR to address this issue was resolved.
